src/components/Auth/ResetPassword.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { useState, useCallback, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Form, Typography, Input, Button, notification } from 'antd';
+import styled from 'styled-components';
+import qs from 'query-string';
+
+import { useAuth } from '../../hooks';
+import { validatePassword } from './Auth';
+
+function ResetPassword() {
+  const [form] = Form.useForm();
+  const { authenticating, onPasswordReset, isPasswordResetted } = useAuth();
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState(' ');
+
+  const history = useHistory();
+
+  const q = qs.parse(document.location.search);
+  const token = q.token as string;
+  const login = (q.email as string) || '';
+
+  const handlePasswordChange = useCallback((e: any) => {
+    setPassword(e.target.value);
+  }, []);
+  const handleconfirmPasswordChange = useCallback((e: any) => {
+    setConfirmPassword(e.target.value);
+  }, []);
+
+  const resetSuccessfulNotification = () => {
+    notification.success({
+      message: 'Password has been successfully changed.',
+      description: 'Your password has been successfully changed. You can log in with your new password now.',
+      duration: 3.5,
+    });
+  };
+
+  const handlePasswordReset = useCallback(async () => {
+    await form.validateFields();
+    if (password === confirmPassword) {
+      onPasswordReset({ login, password, token });
```

Comment:
after you `await` on this then just show the notification in the next line and you will not need `isPasswordResetted` at all